### PR TITLE
LRAC-11981 - [F] Create modal to assign channels and sites to the property

### PR DESCRIPTION
**Pull request by ilzamcmed**

### CODE REVIEW

**Comment by ilzamcmed** [comment]
ci:test:sf

**Comment by ilzamcmed** [comment]
ci:test:relevant

**Comment by ilzamcmed** [comment]
ci:test:analytics-cloud-acceptance

**Comment by ilzamcmed** [comment]
ci:test:sf

**Comment by ilzamcmed** [comment]
ci:test:sf

**Review by fmarcio — APPROVED** [review]
LGTM, thanks @ilzamcmed :rocket:

**Comment by JLucasM** [comment]
LGTM! Good job @ilzamcmed

**Review by JLucasM — APPROVED** [review]
LGTM! Good job @ilzamcmed

**Comment by ilzamcmed** [comment]
ci:foward

**Comment by fmarcio** [comment]
ci:foward

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+					</ClayTabs.Content>
+				</ClayTabs>
+			</ClayModal.Body>
+
+			<ClayModal.Footer
+				last={
+					<ClayButton.Group spaced>
+						<ClayButton
+							displayType="secondary"
+							onClick={onCloseModal}
```
Comment (interaminense) [inline modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx:89]: Use this way to follow the type `() => void`

```suggestion
							onClick={() => onCloseModal()}
```

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import React, {useState} from 'react';
+
+import {TProperty} from '../components/PropertiesTable';
+import ChannelTab from './ChannelTab';
+import SitesTab from './SitesTab';
+
+interface IAssignModal {
+	observer: any;
+	onCloseModal: () => void;
+	propertyData: TProperty;
```
Comment (interaminense) [inline modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx:27]: Could you rename `propertyData` to `property`?

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import ClayButton from '@clayui/button';
+import ClayModal from '@clayui/modal';
+import ClayTabs from '@clayui/tabs';
+import React, {useState} from 'react';
+
+import {TProperty} from '../components/PropertiesTable';
+import ChannelTab from './ChannelTab';
+import SitesTab from './SitesTab';
+
+interface IAssignModal {
```
Comment (interaminense) [inline modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx:24]: Rename `IAssignModal` to `IAssignModalProps`

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	Channel = 0,
+	Sites = 1,
+}
+
+const AssignModal: React.FC<IAssignModal> = ({
+	observer,
+	onCloseModal,
+	propertyData,
+}) => {
+	const [activeTabKeyValue, setActiveTabKeyValue] = useState<number>(
```
Comment (interaminense) [inline modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx:40]:
```suggestion
	const [activeTabKeyValue, setActiveTabKeyValue] = useState<Etabs>(
```

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+	return (
+		<ClayModal observer={observer} size="lg">
+			<ClayModal.Header>
+				{Liferay.Language.get('assign-to')} {propertyData.name}
+			</ClayModal.Header>
+
+			<ClayModal.Body>
+				<ClayTabs modern>
+					<ClayTabs.Item
+						active={activeTabKeyValue === 0}
```
Comment (interaminense) [inline modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx:53]:
```suggestion
						active={activeTabKeyValue === ETabs.Channel}
```

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+						innerProps={{
+							'aria-controls': 'tabpanel-1',
+						}}
+						onClick={() => setActiveTabKeyValue(0)}
```
Comment (interaminense) [inline modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx:57]:
```suggestion
						onClick={() => setActiveTabKeyValue(ETabs.Channel)}
```

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+					>
+						{Liferay.Language.get('channel')}
+					</ClayTabs.Item>
+
+					<ClayTabs.Item
+						active={activeTabKeyValue === 1}
```
Comment (interaminense) [inline modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx:63]:
```suggestion
						active={activeTabKeyValue === ETabs.Sites}
```

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+						innerProps={{
+							'aria-controls': 'tabpanel-2',
+						}}
+						onClick={() => setActiveTabKeyValue(1)}
```
Comment (interaminense) [inline modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx:67]:
```suggestion
						onClick={() => setActiveTabKeyValue(ETabs.Sites)}
```

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/pages/wizard/PropertyStep.tsx [hunk 9]
```diff
@@ -27,15 +27,15 @@ interface IStepProps extends TGenericComponent {}
...
 
 	useEffect(() => {
 		const request = async () => {
 			const response = await fetchProperties();
 			setProperties(response.items);
 		};
 		request();
-	}, []);
-	const {observer, onOpenChange, open} = useModal();
+	}, [properties]);
```
Comment (interaminense) [inline modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/pages/wizard/PropertyStep.tsx:38]: We don't need to add `property` as a dependency if it is not used inside useEffect, you can remove it.

**Review by interaminense — CHANGES_REQUESTED** [review]
Hey @ilzamcmed. Just few comments.

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx (hunk 3 above)
Comment (ilzamcmed) [reply modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx:24]: Fixed

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx (hunk 4 above)
Comment (ilzamcmed) [reply modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx:40]: Fixed

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx (hunk 5 above)
Comment (ilzamcmed) [reply modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx:53]: Fixed

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx (hunk 6 above)
Comment (ilzamcmed) [reply modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx:57]: Fixed

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx (hunk 7 above)
Comment (ilzamcmed) [reply modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx:63]: Fixed

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx (hunk 8 above)
Comment (ilzamcmed) [reply modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/AssignModal.tsx:67]: Fixed

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/pages/wizard/PropertyStep.tsx (hunk 9 above)
Comment (ilzamcmed) [reply modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/pages/wizard/PropertyStep.tsx:38]: If I remove it, the properties list will not be reloaded when I create a new property

Comment (ilzamcmed) [reply modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/pages/wizard/PropertyStep.tsx]: Updated as you suggestion

**Comment by interaminense** [comment]
ci:forward

**Comment by ilzamcmed** [comment]
ci:test:sf

**Review by interaminense — APPROVED** [review]
LGTM! Thanks @ilzamcmed !

**Pull request closed without merging**